validator/client/service.go

```diff
@@ -227,8 +233,47 @@ func (v *ValidatorService) Start() {
 		eventsChannel:                  make(chan *eventClient.Event, 1),
 	}
 
-	v.validator = valStruct
-	go run(v.ctx, v.validator)
+	healthTracker := v.validator.HealthTracker()
+	// Start the health check routine
+	go runHealthCheckRoutine(v.ctx, v.cancel, v.validator, v.MaxHealthChecks)
```

Comment:
```suggestion
	runHealthCheckRoutine(v.ctx, v.cancel, v.validator, v.MaxHealthChecks)
```
There is no point in using `go` here because `runHealthCheckRoutine` is non-blocking